scripts/apps/extension-points/services/ExtensionPoints.js

```diff
@@ -6,27 +6,30 @@
  * @description
  * External superdesk apps can register components that then will be hooked into
  * the core UI.
- * Inject 'extensionPointsProvider' into your module and then:
+ * Inject 'extensionPoints' into your module and then:
  *
- * extensionPointsProvider.register('MY_TYPE', MyComponent, an_array); // do this in config phase
+ * extensionPoints.register('MY_TYPE', MyConnectedComponent, props, data); // do this in config phase
  *
- * where MyComponent is a React component and an_array is an array with names
- * of variables that your component will receive as props from the parent
- * scope, for example: ['item'].
+ * where MyConnectedComponent is a React component, the props is a json object that may contain a
+ * redux store and data is an array with names of variables that your component will receive as props
+ * from the parent scope, for example: ['item'].
  *
  * See also ExtensionPointsDirective.
  */
-export function ExtensionPointsProvider() {
-    var extensions = {};
+export function ExtensionPointsService() {
+    this.extensions = {};
 
-    this.register = function(type, componentClass, data) {
-        if (typeof extensions[type] === 'undefined') {
-            extensions[type] = [];
+    this.register = function(type, componentClass, props = {}, data = []) {
+        if (typeof this.extensions[type] === 'undefined') {
+            this.extensions[type] = [];
         }
-        extensions[type].push({type: type, componentClass: componentClass, data: data});
+        this.extensions[type].push({type, componentClass, props, data});
     };
 
-    this.$get = function() {
-        return extensions;
+    this.get = (type) => {
+        if (typeof this.extensions[type] === 'undefined') {
+            return [];
```

Comment:
It would be nice to let the user know if this happens. Either by `console.error`, or `throw`. Otherwise it can cause hard to debug problems.

Reply:
I think it's the best thing to do to ignore this case silently. Let me try to explain why:
The purpose of this plugin is to make the superdesk UI extendable. For this to work you'd need to add "extension points" (i.e. an angular directive) in all of the places that potentially someone else might want to hook in to. This piece of code is only used by that angular directive, not by the developer that is actually writing an extension. So if we sprinkle extension points all over the UI, probably only some of them are actually used by an extension - maybe even none, depending on the setup. So when there is an extension point that is not used, the above happens. But that's not an error, it is rather the default

Reply:
I see. My reasoning was that you'd want to know in case you potentially mistype the name of the component.